File: 정수빈/LeetCode/medium/insert-interval.py
# https://leetcode.com/problems/insert-interval/description/?envType=study-plan-v2&envId=top-interview-150
# 답지봄 ㅠㅠ

class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        res = []
        
        for i, interval in enumerate(intervals):
            if newInterval[1] < interval[0]:
                res.append(newInterval)
                return res + intervals[i:]
            elif newInterval[0] > interval[1]:
                res.append(interval)
            else:
                newInterval = [min(newInterval[0], interval[0]), max(newInterval[1], interval[1])]
        res.append(newInterval)
        return res    

File: 장세훈/LeetCode/medium/InsertInterval.java
class InsertInterval {
  /*
    https://leetcode.com/problems/insert-interval/
   */
  public int[][] insert(int[][] intervals, int[] newInterval) {
    List<int[]> result = new ArrayList<>();

    boolean added = false;
    for (int[] itv : intervals) {
      if (added || itv[1] < newInterval[0]) {
        result.add(itv);
      } else if (itv[0] > newInterval[1]) {
        result.add(newInterval);
        result.add(itv);
        added = true;
      } else {
        newInterval[0] = Math.min(newInterval[0], itv[0]);
        newInterval[1] = Math.max(newInterval[1], itv[1]);
      }
    }

    if (!added) {
      result.add(newInterval);
    }

    return result.toArray(new int[result.size()][]);
  }
}